### PR TITLE
Update README For Soot's Java 9 Features

### PR DESCRIPTION
**Pull request by SruthiVenkat**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,17 +89,15 @@ Thanks!
...
 ## Use from Source Code
 To load modules in Soot's `ModuleScene` from java:
 ```.java
-// configure Soot's options
+// configure Soot's options, refer to example configurations below
 Options.v().set_soot_modulepath(modulePath);
 
 
 // load classes from modules into Soot
+// Here, getClassUnderModulePath() expects the module path to be set using the Options class
```
Comment (patricklam) [inline README.md:100]: (as seen above)

File: README.md [hunk 2]
```diff
@@ -89,17 +89,15 @@ Thanks!
...
 Map<String, List<String>> map = ModulePathSourceLocator.v().getClassUnderModulePath(modulePath);
 for (String module : map.keySet()) {
    for (String klass : map.get(module)) {
```
Comment (patricklam) [inline README.md:103]: let's explain below where we have the call to loadClass() that it's defined just below

File: README.md [hunk 3]
```diff
@@ -123,31 +121,32 @@ public static SootClass loadClass(String name, boolean main, String module) {
 }
 
 ```
-
+ModuleUtil.module_mode() helps you check if you have modules enabled in Soot. This is done based on whether the module path is set using the Options class.
```
Comment (patricklam) [inline README.md:124]: if -> whether

File: README.md [hunk 4]
```diff
@@ -123,31 +121,32 @@ public static SootClass loadClass(String name, boolean main, String module) {
...
 
 ### Example Configurations: Java 8, Java >= 9 Classpath, Java >= 9 Modulepath
 
 ```.java
 
-if(java < 9 ) {
+if(java < 9 ) { // when you want Java < 9 for Soot and you are not using Soot on modules
```
Comment (patricklam) [inline README.md:130]: Can we clarify? I think "when you have a target benchmark with Java < 9 and hence no modules" here.

File: README.md [hunk 5]
```diff
@@ -123,31 +121,32 @@ public static SootClass loadClass(String name, boolean main, String module) {
...
     Options.v().set_prepend_classpath(true);
     Options.v().set_process_dir(Arrays.asList(applicationClassPath().split(File.pathSeparator)));
-    Options.v().set_claspath(sootClassPath();
+    Options.v().set_soot_classpath(sootClassPath());
 }
 
-if(java >= 9 && USE_CLASSPATH){
+if(java >= 9 && USE_CLASSPATH) { // when you want Java >= 9 for Soot and you are not using Soot on modules
```
Comment (patricklam) [inline README.md:136]: when you have a target benchmark which is for Java >= 9 and want no module support

File: README.md [hunk 6]
```diff
@@ -123,31 +121,32 @@ public static SootClass loadClass(String name, boolean main, String module) {
...
     Options.v().set_soot_classpath("VIRTUAL_FS_FOR_JDK" + File.pathSeparator + sootClassPath());
     Options.v().set_process_dir(Arrays.asList(applicationClassPath().split(File.pathSeparator)));
 }
 
 
-if(java>=9 && USE_MODULEPATH){
+if(java>=9 && USE_MODULEPATH) { // when you want Java >= 9 for Soot and you are using Soot on modules
```
Comment (patricklam) [inline README.md:142]: when you have a target with Java >= 9 and modules.

File: README.md [hunk 7]
```diff
@@ -123,31 +121,32 @@ public static SootClass loadClass(String name, boolean main, String module) {
...
     Options.v().set_prepend_classpath(true);
-    Options.v().set_soot_modulepath(ootClassPath());
+    Options.v().set_soot_modulepath(sootClassPath());
     Options.v().set_process_dir(Arrays.asList(applicationClassPath().split(File.pathSeparator)));
 }
 
 ```
+In the above examples, applicationClassPath() should be replaced with the path to the application classes for analysis by Soot and sootClassPath() should be replaced with Soot's classpath.
```
Comment (patricklam) [inline README.md:149]: Soot's classpath -> the Soot classpath.

**Comment by anddann** [comment]
Thx for the review @patricklam

**Review by anddann — APPROVED** [review]
Thx for the changes, and sorry for the delay

**Comment by anddann** [comment]
@linghuiluo Since you spend more time in the last weeks on Soot, I just wanted to ask you if we can merge this. I'm actually fine with the changes.

**Pull request merged**